### PR TITLE
Refactor use of Informers in UO and SPS controller

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Refactoring

### Description

This PR refactors how we use Informers in our current code base (User Operator and StrimziPodSetController). It creates a new wrapper/encapsulation of the informer and lister to make it easier to share them through the code and query the cache. This should simplify the existing code and make it easier to introduce informers in other parts of the code base.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/kubernetes/AbstractNamespacedResourceOperator.java [hunk 1]
```diff
@@ -403,33 +404,33 @@ public Future<Void> deleteAsync(Reconciliation reconciliation, String namespace,
...
+    // *
+    // * @param namespace         Namespace on which to inform
+    // * @param selectorLabels    Selector which should be matched by the resources
+    // * @param resyncIntervalMs  The interval in which the resync of the informer should happen in milliseconds
+    // *
+    // * @return                  Informer instance
+    // */
+    //public Informer<T> informer(String namespace, Map<String, String> selectorLabels, long resyncIntervalMs)   {
+    //    return new Informer<>(runnableInformer(applyNamespace(namespace).withLabels(selectorLabels), resyncIntervalMs));
+    //}
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/kubernetes/AbstractNamespacedResourceOperator.java:433]: can we remove all the above commented code?

**Review by ppatierno — APPROVED** [review]
LGTM. I left just a nit.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/kubernetes/AbstractNamespacedResourceOperator.java (hunk 1 above)
Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/kubernetes/AbstractNamespacedResourceOperator.java]: Damn, I thought I deleted all of them. Thanks for catching this.

**Pull request merged**